intermediate_source/inductor_debug_cpu.py

```diff
@@ -0,0 +1,600 @@
+#     inductor use: 339.95180135127157 ms/iter
+#     speed up ratio: 2.359459053287382
+#
+# The inductor model speed-up is 2.58x.
+#
+#
+# Secondly, we can deep dive into op-level performance to understand where is the speed-up comes from.
+# `Pytorch Profiler <https://pytorch.org/tutorials/recipes/recipes/profiler_recipe.html>`_ is a good tool to help us.
+# To enable kernel profile with inductor model, we need to set ``enable_kernel_profile`` by:
+
+from torch._inductor import config
+config.cpp.enable_kernel_profile = True
+
+######################################################################
+# Following the steps in `Pytorch Profiler <https://pytorch.org/tutorials/recipes/recipes/profiler_recipe.html>`_
+# We are able to get the profiling table and trace files.
+
+from torch.profiler import profile, schedule, ProfilerActivity
+RESULT_DIR = "./prof_trace"
+my_schedule = schedule(
+    skip_first=10,
+    wait=5,
+    warmup=5,
+    active=1,
+    repeat=5)
+
+def trace_handler(p):
+    output = p.key_averages().table(sort_by="self_cpu_time_total", row_limit=20)
+    print(output)
+    p.export_chrome_trace(f"{RESULT_DIR}/{p.step_num}.json")
+
+for _ in range(10):
+    model(**input_dict)  # inductor_model(**input_dict) to get inductor model profiling
+
+total = 0
+with profile(
+    activities=[ProfilerActivity.CPU],
+    schedule=my_schedule,
+    on_trace_ready=trace_handler
+) as p:
+    for _ in range(100):
+        model(**input_dict)  # inductor_model(**input_dict) to get inductor model profiling
+        p.step()
+
+######################################################################
+# We will get the following profile table for the eager model:
+#
+# .. image:: ../_static/img/eager_prof.png
+#
+# Similarly, get the table for the inductor model:
+#
+# .. image:: ../_static/img/inductor_prof.png
+#
+# From the profiling table of the eager model, we can see the most time consumption ops are [aten::addmm, aten::add, aten::copy_, aten::mul, aten::clamp_min, aten::bmm].
+# Comparing with the inductor model profiling table, we notice there are ``mkl::_mkl_linear`` and fused kernel called ``graph_0_cpp_fused_*``. They are the major
+# optimization that the inductor model is doing. Let us discuss them separately.
+#
+# (1) Regard to ``mkl::_mkl_linear``: You may notice the number of calls to this kernel is 362, which is exactly the same as ``aten::linear`` in the eager model profiling table.
+# The CPU total of ``aten::linear`` is 376.888ms, at the mean time it is 231.573ms for ``mkl::_mkl_linear``. This suggests inductor model speed up ~1.63x for the "linear" part.
+#
+# (2) Regarding non-linear part: The end-to-end latency for the eager/inductor model is 802/339ms. The speed up for the non-linear part is ~3.94x.
+# Let's read the generated code to understand how the inductor achieves this impressive optimization. You are able to find the generated code by 
+# searching ``cpp_fused__mkl_linear_add_mul_relu_151`` in ``output_code.py``
```

Comment:
Yes, I checked how the kernel name is generated. It seems the kernel name is generated by all ops in a `scheduler node`. And since `mkl_linear` is not an external call: it might be contained by a scheduler node. I am afraid we do not have enough time to fix it first.